### PR TITLE
feat: support auto create mcs with flags

### PR DESCRIPTION
**Pull request by duanmengkk**
#### What type of PR is this?

/kind feature

#### Which issue(s) does this PR fix?

Fixes #
#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: pkg/clustertree/cluster-manager/controllers/mcs/auto_mcs_controller.go [hunk 1]
```diff
@@ -124,15 +134,19 @@ func (c *AutoCreateMCSController) SetupWithManager(mgr manager.Manager) error {
...
-			return false
+
+		annotations := obj.GetAnnotations()
+		if annotations != nil {
+			if _, exists := annotations[utils.AutoCreateMCSAnnotation]; exists {
+				return true
+			}
 		}
+		return false
 	}
```
Comment (OrangeBao) [inline pkg/clustertree/cluster-manager/controllers/mcs/auto_mcs_controller.go]: line 142-line149  can reuse the func ```hasAutoMCSAnnotation```  
use ```obj.(*corev1.Service)``` to convert client.Object to corev1.Service

File: cmd/clustertree/cluster-manager/app/options/options.go [hunk 2]
```diff
@@ -70,6 +73,6 @@ func (o *Options) AddFlags(flags *pflag.FlagSet) {
 	flags.BoolVar(&o.MultiClusterService, "multi-cluster-service", false, "Turn on or off mcs support.")
 	flags.StringVar(&o.RootCoreDNSServiceNamespace, "root-coredns-service-namespace", CoreDNSServiceNamespace, "The namespace of the CoreDNS service in the root cluster, used to locate the CoreDNS service when MultiClusterService is disabled.")
 	flags.StringVar(&o.RootCoreDNSServiceName, "root-coredns-service-name", CoreDNSServiceName, "The name of the CoreDNS service in the root cluster, used to locate the CoreDNS service when MultiClusterService is disabled.")
-
+	flags.StringSliceVar(&o.AutoCreateMCSPrefix, "auto-mcs-prefix", []string{}, "The prefix of namespace of service to auto create mcs resources")
```
Comment (wuyingjun-lucky) [inline cmd/clustertree/cluster-manager/app/options/options.go]: The prefix of namespace for service to auto create mcs resources

File: pkg/clustertree/cluster-manager/controllers/mcs/auto_mcs_controller.go [hunk 3]
```diff
@@ -84,6 +83,41 @@ func (c *AutoCreateMCSController) Reconcile(ctx context.Context, request reconci
 	return controllerruntime.Result{}, nil
 }
 
+func namespaceMatch(namespace string, prefix []string) bool {
```
Comment (wuyingjun-lucky) [inline pkg/clustertree/cluster-manager/controllers/mcs/auto_mcs_controller.go]: matchNamespace

**Pull request merged**